File: ios/JJMproject/Classes/JSBridge/Modules/Location/LocationModule.m
//
//  NetworkModule.m
//  MarioFramework
//
//  Created by MeePwn on 2017/11/15.
//  Copyright © 2017年 Facebook. All rights reserved.
//

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(LocationModule, NSObject)

RCT_EXTERN_METHOD(requestLocationInfo:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(stopLocation)

@end

File: ios/JJMproject/Classes/JSBridge/Modules/Location/LocationModule.swift
//
//  LocationModule.swift
//  MarioFramework
//
//  Created by MeePwn on 2017/11/20.
//  Copyright © 2017年 Facebook. All rights reserved.
//

import UIKit
import Async

@objc(LocationModule)
class LocationModule: NSObject {
  
  private lazy var locationUtility: LocationUtility = {
    return LocationUtility()
  }()
  
  @objc func requestLocationInfo(_ resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
    Async.main {
      self.locationUtility.requestLocationInfo(block: { (locationInfo, error) in
        if error != nil {
          reject(ErrorCode.LOCATION_FAILURE.rawValue, "定位失败", error)
        } else {
          resolve(locationInfo)
        }
      })
    }
  }
  
  @objc func stopLocation() {
    Async.main {
      self.locationUtility.stopLocation()
    }
  }

}
